htcanalyze/main.py

```diff
@@ -565,112 +514,125 @@ def print_results(htcanalyze: HTCAnalyze,
-                    del mystery["all-resources"]["Usage"]
+                    del data_dict["all-resources"]["Usage"]
                 if "requested-resources" in ignore_list:
-                    del mystery["all-resources"]["Requested"]
+                    del data_dict["all-resources"]["Requested"]
                 if "allocated-resources" in ignore_list:
-                    del mystery["all-resources"]["Allocated"]
+                    del data_dict["all-resources"]["Allocated"]
 
-                table = wrap_dict_to_table(mystery["all-resources"])
+                table = wrap_dict_to_table(data_dict["all-resources"])
                 rprint(table)
 
-        if "ram-history" in mystery:
+        if "ram-history" in data_dict:
             if "ram-history" in ignore_list:
-                del mystery["ram-history"]
-            elif mystery["ram-history"] is not None:
-                print(mystery["ram-history"])
+                del data_dict["ram-history"]
+            elif data_dict["ram-history"] is not None:
+                print(data_dict["ram-history"])
 
-        if "errors" in mystery:
+        if "errors" in data_dict:
             if "errors" in ignore_list:
-                del mystery["errors"]
-            elif mystery["errors"] is not None:
-                table = wrap_dict_to_table(mystery["errors"],
+                del data_dict["errors"]
+            elif data_dict["errors"] is not None:
+                table = wrap_dict_to_table(data_dict["errors"],
                                            "Occurred HTCondor errors")
                 rprint(table)
 
-        if "host-nodes" in mystery:
+        if "host-nodes" in data_dict:
             if "host-nodes" in ignore_list:
-                del mystery["host-nodes"]
-            elif mystery["host-nodes"] is not None:
-                table = wrap_dict_to_table(mystery["host-nodes"])
+                del data_dict["host-nodes"]
+            elif data_dict["host-nodes"] is not None:
+                table = wrap_dict_to_table(data_dict["host-nodes"])
                 rprint(table)
 
         # Show more section
-        if "htc-out" in mystery and mystery["htc-out"] != "":
+        if "htc-out" in data_dict and data_dict["htc-out"] != "":
             rprint("\n[bold cyan]Related HTC standard output:[/bold cyan]")
-            rprint(mystery["htc-out"])
+            rprint(data_dict["htc-out"])
 
-        if "htc-err" in mystery and mystery["htc-err"] != "":
+        if "htc-err" in data_dict and data_dict["htc-err"] != "":
             rprint("\n[bold cyan]Related HTCondor standard error:[/bold cyan]")
-            rprint(mystery["htc-err"])
+            rprint(data_dict["htc-err"])
 
         print()
 
 
+def check_for_redirection() -> (bool, bool, list):
+    """Check if reading from stdin or redirecting stdout."""
+    redirecting_stdout = not sys.stdout.isatty()
+    reading_stdin = not sys.stdin.isatty()
+    stdin_input = None
+
+    if reading_stdin:
```

Comment:
`stdin_input = sys.stdin.readlines() if reading_stdin else None`

Then you can also remove `stdin_input = None`.